core/src/main/scala/frameless/CatalystOrdered.scala

```diff
@@ -27,4 +28,21 @@ object CatalystOrdered {
       i0: Injection[A, B],
       i1: CatalystOrdered[B]
     ): CatalystOrdered[A] = of[A]
+
+  implicit def deriveHNil[H](implicit head: CatalystOrdered[H]): CatalystOrdered[H :: HNil] =
+    of[H :: HNil]
+
+  implicit def deriveCons[H, T <: HList]
+    (implicit
+      i0: CatalystOrdered[H],
+      i1: CatalystOrdered[T]
+    ): CatalystOrdered[H :: T] =
+      of[H :: T]
+
+  implicit def deriveCaseClass[C, H <: HList]
+    (implicit
+      i0: Generic.Aux[C, H],
+      i1: Lazy[CatalystOrdered[H]]
```

Comment:
What do you think about using `shapeless.ops.hlist.LiftAll[CatalystOrdered, H]` instead?

Reply:
@OlivierBlanvillain: Thanks, didn't know about `LiftAll`, looks much neater this way.

I did end up removing `deriveHCons`/`deriveHList` as:
  - they weren't needed anymore (for case classes/tuples)
  - they weren't specifically tested
  - I don't know if deriving a `CatalystOrdered[HList]` would be all that useful (plus it's not what the issue was about).

I can add back a `deriveHList` (using `LiftAll` too) if you think it'd be worth it.